cucumber-selenium-tests/src/test/java/com/skillstorm/StepDefinitions/GenericStepDefinitions.java

```diff
@@ -1,19 +1,88 @@
 package com.skillstorm.StepDefinitions;
 
+import io.cucumber.java.en.*;
 
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.Select;
+import org.openqa.selenium.support.ui.WebDriverWait;
 
-public class GenericStepDefinitions {
+import com.skillstorm.Utilities.Navigator;
 
-    // @Given("an example scenario")
-    // public void anExampleScenario() {
-    // }
 
-    // @When("all step definitions are implemented")
-    // public void allStepDefinitionsAreImplemented() {
-    // }
+//This class implements some step definitions that are commonly used among our tests. You must pass in the web driver you're using as a parameter.
+public class GenericStepDefinitions{
 
-    // @Then("the scenario passes")
-    // public void theScenarioPasses() {
-    // }
+    WebDriver driver;
+    WebDriverWait wait;
+    Navigator navigator = new Navigator(driver);
 
+    public GenericStepDefinitions(WebDriver driver) {
```

Comment:
Since this is the actual step definitions and not a "class" should this constructor actually be a @BeforeSuite or @BeforeClass method that initiates the the "driver", "wait", and "navigator" variables before all the tests start? Anyone else can also chime in on this, its just my understanding of how Cucumber works

Reply:
adding here that maybe the given I am on page step definition could also load the "page" object (Noted its use down below in the @When I click a button section

Reply:
Atleast for my experience with Project 2, we didn't have constructors within our step definitions files. Our setup was solely inside a before() method with @Before attached to it. That's where we'd have:

ChromeOptions options = new ChromeOptions();
        options.setBrowserVersion("127");
        options.addArguments("--headless", "--disable-dev-shm-usage", "--ignore-ssl-errors=yes", "'--ignore-certificate-errors'");
        this.driver = new ChromeDriver(options);
        this.navigator= new Navigator(driver);